unit-tests/src/test/java/org/eclipse/collections/impl/map/mutable/MutableMapIterableTestCase.java

```diff
@@ -560,6 +563,18 @@ public void withKeyValue()
         Verify.assertMapsEqual(UnifiedMap.newWithKeysValues("A", 11, "B", 2), mapWith);
     }
 
+    @Test
+    public void withMap()
+    {
+        MutableMapIterable<String, Integer> map = this.newMapWithKeysValues("A", 1, "B", 2);
```

Comment:
Can you add a test case for empty maps?
1) Input map is empty, target map is non-empty
2) Input map is empty, target map is empty
3) Input map is non-empty, target map is empty

Reply:
I added them!